Products/ZenWidgets/browser/Portlets.py

```diff
@@ -88,19 +89,26 @@ def getDevProdStateJSON(self, prodStates=['Maintenance']):
                 {'Device':'<a href=/>', 'Prod State':'Maintenance'},
             ]}"
         """
-        devroot = self.context.dmd.Devices
         if isinstance(prodStates, basestring):
             prodStates = [prodStates]
 
+        def getProdStateInt(prodStateString):
+            for t in self.context.getProdStateConversions():
+                if t[0] == prodStateString:
+                    return t[1]
+
+        numericProdStates = [getProdStateInt(p) for p in prodStates]
+
         orderby, orderdir = 'id', 'asc'
-        catalog = getattr(devroot, devroot.default_catalog)
-        query = Indexed('id') # Get all the brains
+        catalog = self.context.getPhysicalRoot().zport.global_catalog
```

Comment:
Other way to get the global catalog:
from Products.Zuul.catalog.interfaces import IModelCatalogTool
global_catalog = IModelCatalogTool(self.context)